### PR TITLE
Added submission id to url

### PR DESCRIPTION
**Pull request by Yalikesifulei**
Now it's possible to specify submission id in url by adding `?id=30512132`, so links directly to some submission stats can be shared

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -29,7 +31,8 @@
     st.title("Lux AI submission statistics")
     col1, col2 = st.columns(2)
 
-    sub_id = col1.number_input("Submission ID", value=30131764, step=1, key="sub_id")
+    sub_id = col1.number_input("Submission ID", value=int(query_params['id'][0]), step=1, key="sub_id")
```
Comment (andrii0yerko) [inline main.py:34]: An empty url query leads to a crash now. Consider fallback to default submission id
```
DEFAULT_SUB_ID = 30512132
sub_id = col1.number_input("Submission ID", value=int(query_params.get("id", [DEFAULT_SUB_ID])[0]), step=1, key="sub_id")
```

File: main.py [hunk 2]
```diff
@@ -29,7 +31,8 @@
     st.title("Lux AI submission statistics")
     col1, col2 = st.columns(2)
 
-    sub_id = col1.number_input("Submission ID", value=30131764, step=1, key="sub_id")
+    sub_id = col1.number_input("Submission ID", value=int(query_params['id'][0]), step=1, key="sub_id")
+    st.experimental_set_query_params(**{'id': str(sub_id)})
```
Comment (andrii0yerko) [inline main.py:35]: "id"

:smile:

**Review by andrii0yerko — CHANGES_REQUESTED** [review]
Also, I believe there are some problems with this `experimental_get_query_params` method on `streamlit==1.1.0` that are currently frozen in requirements ("Unable to fetch data"), while `streamlit==1.19.0` works nicely

I guess we should consider updating dependencies versions but need to test it won't break python 3.8 deployments (I'm testing with p3.10 locally)

File: main.py (hunk 2 above)
Comment (Yalikesifulei) [reply main.py]: Done

**Comment by andrii0yerko** [comment]
Maybe we can load stats right away if url query was parsed? Instead of only filling out the form value

**Comment by andrii0yerko** [comment]
My suggestion is to restructure our main:

1. move lines 39 - 71 to separate the function of sub_id: `def display_stats(sub_id)`
2. Restructure `if __name__=="__main__"` block. Let it consists of three blocks:
    1. Setup: set_page_config/title/ etc
    2. Check for url query. If appears - call `display_stats` directly
    3. if the button was clicked  - call `display_stats` as before

**Comment by andrii0yerko** [comment]
Also now incrementing or decrementing of a form value leads to stats loading immediately - a very unpleasant behavior considering its speed

**Comment by andrii0yerko** [comment]
Just have investigated it with @Yalikesifulei 

The reason was streamlit flushes of the global scope on reruns. [`st.session_state`](https://docs.streamlit.io/library/api-reference/session-state) fixed the problem.

**Review by andrii0yerko — APPROVED** [review]

**Comment by andrii0yerko** [comment]
Approved, thanks for the contribution

**Pull request merged**